app/src/docs/_examples/tables/PagedTableWithSelectAll.example.tsx

```diff
@@ -0,0 +1,100 @@
+import React, { useCallback, useMemo, useState } from 'react';
+import { DataColumnProps, useLazyDataSource, DataSourceState, LazyDataSourceApiRequest, useUuiContext, LazyDataSourceApi, ICheckable } from '@epam/uui-core';
+import { DataTable, Panel, Text, Paginator, FlexRow, FlexSpacer } from '@epam/uui';
+import { Person } from '@epam/uui-docs';
+import css from './TablesExamples.module.scss';
+
+export default function PagedTableWithSelectAll() {
+    const svc = useUuiContext();
+    const [state, setState] = useState<DataSourceState>({
+        page: 1, pageSize: 5,
+    });
+
+    const columns: DataColumnProps<Person>[] = useMemo(
+        () => [
+            {
+                key: 'name',
+                caption: 'Name',
+                render: (person) => (
+                    <Text color="primary" fontWeight="600">
+                        {person.name}
+                    </Text>
+                ),
+                isSortable: true,
+                grow: 1,
+                width: 224,
+            }, {
+                key: 'location',
+                caption: 'Location',
+                render: (person) => <Text>{person.locationName}</Text>,
+                width: 144,
+            },
+        ],
+        [],
+    );
+
+    const api: LazyDataSourceApi<Person, number, unknown> = useCallback(
+        async (rq: LazyDataSourceApiRequest<{}>) => {
+            const result = await svc.api.demo.personsPaged({
+                ...rq,
+                filter: { departmentId: 13 }, // to get less results and non round-numbered number of people
+            });
+            return result;
+        },
+        [svc.api.demo],
+    );
+
+    const selectAll = useCallback(async (shouldSelectAll: boolean) => {
```

Comment:
Actually, this work-around looks completely legit and straightforward solution to the problem.